money.go

```diff
@@ -107,53 +166,80 @@ func (m *Money) Negative() *Money {
 		Dollars:        nm.AsMajorUnits(),
 		CurrencyIso:    m.CurrencyIso,
 		CurrencySymbol: m.CurrencySymbol,
+		roundingMode:   m.roundingMode,
 	}
 }
 
 // Add returns new Money struct with value representing sum of Self and Other Money.
-func (m *Money) Add(om *Money) (*Money, error) {
-	m.initMoney()
-	om.initMoney()
-
-	nm, err := m.money.Add(om.money)
-	if err != nil {
-		return nil, err
+func (m *Money) Add(oms ...*Money) (*Money, error) {
```

Comment:
can refactor add like this. just reuse the embedded money object to do add operation and wrap it in the return
```go
func (m *Money) Add(oms ...*Money) (*Money, error) {
	m.initMoney()
	innerMoney := m.money
	var err error
	for _, om := range oms {
		om.initMoney()
		innerMoney, err = innerMoney.Add(om.money)
		if err != nil {
			return nil, err
		}
	}
return &Money{
	Cents:          innerMoney.Amount(),
	Dollars:        innerMoney.AsMajorUnits(),
	CurrencyIso:    m.CurrencyIso,
	CurrencySymbol: m.CurrencySymbol,
	roundingMode:   m.getRoundingModeAmongMoneys(oms),
}, nil
```